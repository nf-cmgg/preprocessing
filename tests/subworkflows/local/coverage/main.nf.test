nextflow_workflow {

    name "Test Workflow COVERAGE"
    script "subworkflows/local/coverage/main.nf"
    workflow "COVERAGE"

    tag "subworkflows"
    tag "subworkflows/local"
    tag "subworkflows/local/coverage"

    test("Coverage") {

        when {
            workflow {
                """
                // ch_meta_cram_crai_fasta_fai_roi
                input[0] = Channel.of([
                    [id: "test", single_end: false], // meta
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/cram/sample1.sorted.cram", checkIfExists: true),
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/cram/sample1.sorted.cram.crai", checkIfExists: true),
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna", checkIfExists: true),
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna.fai", checkIfExists: true),
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/regions/roi_chr21.bed", checkIfExists: true),
                ])
                // genelists
                input[1] = Channel.of([
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/regions/genelists/genelist_chr21_per_exon.bed",checkIfExists:true)
                ])
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
