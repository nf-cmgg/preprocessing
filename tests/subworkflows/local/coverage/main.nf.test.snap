{
    "Coverage": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.mosdepth.summary.txt:md5,c929389c608f49ca01d800fb5cc94bb9"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.mosdepth.global.dist.txt:md5,ceec5e216dac6a4e15b961713ee8b16c"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.mosdepth.region.dist.txt:md5,baffaa91e753347fd44c2e6e0a618d1f"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.txt:md5,2d81e108bf4175f2b892ab6e749fdf92"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_genelist_chr21_per_exon.mosdepth.region.dist.txt:md5,e5c7b4f381721888249c57aa55be2d34"
                    ]
                ],
                "5": [
                    "versions.yml:md5,240240886d2b7ad84a62b01cb04c382d",
                    "versions.yml:md5,2d22ebfca674911d28ac60f352a98b1b",
                    "versions.yml:md5,731a006ffa265ac74ad677b4e5a68640"
                ],
                "mosdepth_global": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.mosdepth.global.dist.txt:md5,ceec5e216dac6a4e15b961713ee8b16c"
                    ]
                ],
                "mosdepth_regions": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.mosdepth.region.dist.txt:md5,baffaa91e753347fd44c2e6e0a618d1f"
                    ]
                ],
                "mosdepth_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.mosdepth.summary.txt:md5,c929389c608f49ca01d800fb5cc94bb9"
                    ]
                ],
                "panelcoverage": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_genelist_chr21_per_exon.mosdepth.region.dist.txt:md5,e5c7b4f381721888249c57aa55be2d34"
                    ]
                ],
                "samtools_coverage": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.txt:md5,2d81e108bf4175f2b892ab6e749fdf92"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,240240886d2b7ad84a62b01cb04c382d",
                    "versions.yml:md5,2d22ebfca674911d28ac60f352a98b1b",
                    "versions.yml:md5,731a006ffa265ac74ad677b4e5a68640"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-16T08:42:51.371734"
    }
}